src/compositions/useLink.ts

```diff
@@ -106,24 +119,43 @@ export function useLink(
       routerPrefetch,
       linkPrefetch,
     })
+
+    props = prefetchPropsForRoute(route, {
+      getRouteProps,
+      routerPrefetch,
+      linkPrefetch,
+    })
   }, { immediate: true })
 
+  const push: UseLink['push'] = (options) => {
+    if (props) {
+      setRouteProps(props, { prefetched: true })
+    }
+
+    return router.push(href.value, { ...optionsRef.value, ...options })
+  }
+
+  const replace: UseLink['replace'] = (options) => {
+    return push({ ...options, replace: true })
+  }
+
   return {
     route,
     href,
     isMatch,
     isExactMatch,
-    push: (options?: RouterPushOptions) => router.push(href.value, {}, { ...optionsRef.value, ...options }),
-    replace: (options?: RouterReplaceOptions) => router.replace(href.value, {}, { ...optionsRef.value, ...options }),
+    isExternal,
+    push,
+    replace,
   }
 }
 
-type ComponentPrefect = {
+type ComponentPrefetchContext = {
```

Comment:
this type is almost identical to PrefetchConfigs in types/prefetch.ts. Do we need both types? Even if we need it, we should move this to prefetch.ts and have the types build off each other